app/lib/package/backend.dart

```diff
@@ -123,11 +123,21 @@ class PackageBackend {
         .map((p) => p.name!);
   }
 
-  /// Retrieves package versions ordered by their latest version date.
+  /// Retrieves package versions ordered by their published date descending.
   Future<List<PackageVersion>> latestPackageVersions(
-      {int? offset, int? limit}) async {
-    final pkgPage = await latestPackages(offset: offset, limit: limit);
-    return lookupLatestVersions(pkgPage.packages);
+      {int offset = 0, int limit = 10}) async {
```

Comment:
I would make the limit required - I don't think there is any reasonable default.
```suggestion
      {int offset = 0, required int limit}) async {
```